docs/software/ml/pytorch.md

```diff
@@ -0,0 +1,409 @@
+[](){#ref-uenv-pytorch}
+# PyTorch
+
+The PyTorch software stack was designed with the intention of being able to run
+[Megatron-LM](https://github.com/NVIDIA/Megatron-LM)-based pre-training
+workloads out of the box. Thus, it comes with batteries included and does not
+just provide the bare [PyTorch framework](https://github.com/pytorch/pytorch).
+
+
+!!! note "uenv"
+
+    [PyTorch][ref-uenv-pytorch] is provided via [uenv][ref-uenv].
+    Please have a look at the [uenv documentation][ref-uenv] for more information about uenvs and how to use them.
+
+## Versioning
+
+The PyTorch uenv is versioned according to the PyTorch version it provides.
+
+| version   | node types | system                  |
+|-----------|------------|-------------------------|
+| v2.6.0     | gh200      | clariden, daint         |
+
+=== "v2.6.0"
+
+    ??? info "non-Python packages exposed via the `default` view"
+
+        | Package             | Version          |
+        |---------------------|------------------|
+        | abseil-cpp | 20240722.0 |
```

Comment:
No need to change anything now, but we should try to unify how we list available packages...

Perhaps a simple `spack find` would be easiest for copy pasting? Currently I think everyone formats these things (manually?) into slightly different lists/tables.

CC @RMeli @bcumming

Reply:
I have uses spack find (with a custom formatter). Yes it is probably nice to have everything the same. I prefer a table, though, so I'll leave it like that for now.

Reply:
```bash
uenv start pytorch/v2.6.0:v1 --view=spack
git clone --depth=1 --branch=$UENV_SPACK_REF https://github.com/spack/spack.git
./spack/bin/spack -C /user-environment/config find --format '|____{name}____|____{version}____|' %gcc@13 | tr -s ' ' '\n'
```
needed some cleanup (some duplicates and replacing the underscores with whitespace)

Reply:
@msimberg maybe something for the style guide?

Reply:
We could coordinate these things, however uenv can be a bit unique:
* some are purely built on spack packages from the spack/builtin/repo repo
* some use packages from a custom repo
* some install software after the fact using a post-install script

A one size fits all solution for generaring the list isn't possible, and I would rather leave it up to the uenv authors.